### ISSUE TITLE
Webpack resolver: Support webpack.config.babel.js filename

### ISSUE THREAD

**Issue by kentcdodds** [benmosher/eslint-plugin-import#194]
If you rename your file to `webpack.config.babel` then [webpack will transpile it](http://stackoverflow.com/a/31906902/971592) with babel on the fly.

However doing this makes the webpack resolver fail to load the webpack config.

Right now, my workaround is:

```javascript
// webpack.config.js
/*
 * This file is here until this gets supported:
 */
require('babel-register')
module.exports = require('./webpack.config.babel').default
```

Would love this to be supported in core. But looking at the code I wasn't certain how you'd want it implemented. I was thinking of extracting [this](https://github.com/benmosher/eslint-plugin-import/blob/4aa17e620e3d1bba985b2e1a70926bec86264185/resolvers/webpack/index.js#L31-L46) into a function. But feedback is appreciated.

Thanks!

**Comment by benmosher** [comment]
Makes sense. That is cool, didn't realize that was possible. 

Reminds me of #182. I'd like to avoid slowly rewriting Webpack. This seems like a simple enough thing to fix without biting off integration with enhanced-resolve, though. 

Does #164 seem like a workable solution? I didn't really understand what it was doing until reading the SO post you linked.

**Comment by kentcdodds** [comment]
Looks like #164 is using `interpret`, but not to load the config file (which is what concerns this issue). I've never used `interpret` myself.

**Comment by benmosher** [comment]
Hmm, I read it as "if the config file has an extension that `interpret` understands, register the compiler for all files with that extension". Which makes sense, no knowing what other non-JS files the config might require. 

I've not used it either, but I can imagine it working in such a way. 

I'll try it out, and see if I can add some tests. 

Or if you've got a different idea you want to try, I'm up for that. Sounds like `interpret` is a good place to start, though.

**Comment by gausie** [comment]
`interpret` is what webpack uses to do it!

**Comment by benmosher** [comment]
I believe this was fixed by #164.
